src/main/java/com/google/cloud/spanner/pgadapter/statements/CopyStatement.java

```diff
@@ -0,0 +1,267 @@
+    }
+    if (this.quoteChar != '\0') {
+      format = format.withQuote(this.quoteChar);
+    }
+    format = format.withHeader(this.tableColumns.keySet().toArray(new String[0]));
+    return format;
+  }
+
+  public String getTableName() {
+    return this.tableName;
+  }
+
+  protected void setTableName(String tableName) {
+    this.tableName = tableName;
+  }
+
+  /** @return List of column names specified in COPY statement, if provided. */
+  public List<String> getCopyColumnNames() {
+    return this.copyColumnNames;
+  }
+
+  protected void setCopyColumnNames(List<String> copyColumnNames) {
+    this.copyColumnNames = copyColumnNames;
+  }
+
+  /** @return Format type specified in COPY statement, if provided. */
+  public String getFormatType() {
+    return this.formatType;
+  }
+
+  protected void setFormatType(String formatType) {
+    this.formatType = formatType;
+  }
+
+  /** @return Null string specified in COPY statement, if provided. */
+  public String getNullString() {
+    return this.nullString;
+  }
+
+  protected void setNullString(String nullString) {
+    this.nullString = nullString;
+  }
+
+  /** @return Delimiter character specified in COPY statement, if provided. */
+  public char getDelimiterChar() {
+    return this.delimiterChar;
+  }
+
+  protected void setDelimiterChar(char delimiterChar) {
+    this.delimiterChar = delimiterChar;
+  }
+
+  /** @return Escape character specified in COPY statement, if provided. */
+  public char getEscapeChar() {
+    return this.escapeChar;
+  }
+
+  protected void setEscapeChar(char escapeChar) {
+    this.escapeChar = escapeChar;
+  }
+
+  /** @return Quote character specified in COPY statement, if provided. */
+  public char getQuoteChar() {
+    return this.quoteChar;
+  }
+
+  protected void setQuoteChar(char quoteChar) {
+    this.quoteChar = quoteChar;
+  }
+
+  /** @return True if copy data contains a header, false otherwise. */
+  public boolean hasHeader() {
+    return this.hasHeader;
+  }
+
+  protected void setHasHeader(boolean hasHeader) {
+    this.hasHeader = hasHeader;
+  }
+
+  public int getFormatCode() {
+    return this.formatCode;
+  }
+
+  protected void setFormatCode(int columnCount) {
+    this.formatCode = columnCount;
+  }
+
+  public MutationBuilder getMutationBuilder() {
+    return this.mutationBuilder;
+  }
+
+  protected void setMutationBuilder(MutationBuilder mutationBuilder) {
+    this.mutationBuilder = mutationBuilder;
+  }
+
+  private void verifyCopyColumns() throws SQLException {
+    if (this.copyColumnNames.size() > this.tableColumns.size()) {
+      throw new SQLException("Number of copy columns provided exceed table column count");
+    }
+    LinkedHashMap<String, TypeCode> tempTableColumns = new LinkedHashMap<>();
+    // Verify that every copy column given is a valid table column name
+    for (String copyColumn : this.copyColumnNames) {
+      if (!this.tableColumns.containsKey(copyColumn)) {
```

Comment:
The column names are taken from the query to information schema.  This will return case sensitive names.

Reply:
Sorry, I should have been a little more specific: My worry regards the `this.copyColumnNames`. As far as I understand, those come from the `COPY` statement.

1. Assume that the user is trying to execute the following statement:
```sql
COPY Singers (SingerId, LastName) FROM 'some_file.csv'
```
2. The actual column names in the information schema are `singerid` and `lastname`

I think that would fail in this case.

Having this as a limitation (at least in this version) is probably OK, but we should include a hint in the error message if we do.